modules/system-info/reporters/server.php

```diff
@@ -410,4 +406,56 @@ public function get_elementor_library() {
 			'value' => 'Connected',
 		];
 	}
+
+	/**
+	 * @return array []{read: bool, write: bool, execute: bool}
+	 */
+	public function get_system_paths_permissions() : array {
+		$paths = $this->get_system_paths();
```

Comment:
Maybe pass the exactly paths you need to check instead of checking all system paths and then filter it.
```suggestion
	public function get_system_paths_permissions( $paths ) : array {
```

Reply:
I did this in your way the first time, @StyleShit suggested I will do it like the current solution.

Reply:
Every path check is a Disk Access so, we need to reduce it as much it's possible